applications/structure/bar/application.h

```diff
@@ -250,9 +251,9 @@ class Application : public ApplicationBase<dim, Number>
 
     this->param.update_preconditioner                  = true;
     this->param.update_preconditioner_every_time_steps = 1;
-    this->param.update_preconditioner_every_newton_iterations =
-      this->param.newton_solver_data.max_iter;
-    this->param.update_preconditioner_once_newton_converged = true;
+    this->param.update_preconditioner_every_nonlinear_iterations =
```

Comment:
I renamed also the nonlinear solver variables in the structure module - maybe someone wants to implement some other nonlinear solution strategies in the future?